app/models/manageiq/providers/cloud_manager/provision_workflow.rb

```diff
@@ -141,4 +142,18 @@ def provider_or_tenant_object
     obj = src[:cloud_tenant] || src[:ems]
     load_ar_obj(obj)
   end
+
+  def openstack?(networks)
+    networks = networks.select do |cloud_network|
+      cloud_network.class == ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork::Public ||
+        cloud_network.class == ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork::Private
+    end
+    networks.any?
```

Comment:
Few comments on this part of code:

Filter if the CloudNetwork is or is not OpenStack network can be done also by lookup for _ancestors_ of the class (not pushing for change, it's just my preference, example: ```cloud_network.class.ancestors.include? "ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork"```)

A bit high level comment, but in this change we expect that if network is from Openstack , then it has a method eligible_for_vm_provisioning, which might not be clear in future. I'd suggest adding eligible_for_vm_provisioning method also to CloudNetwork.

Nit: We don't need ```networks``` local variable here, it should be enough ask for any? just after the block end.

Nit2: It would be great not use openstack keyword, but have a generic code (e.g. by adding eligible_for_vm_provisioning method to CloudNetwork model).